### PR TITLE
add condition for remote resources.

### PR DESCRIPTION
**Pull request by morvencao**

### CODE REVIEW

File: install/kubernetes/helm/istio/templates/clusterrole.yaml [hunk 1]
```diff
@@ -1,3 +1,4 @@
+{{- if .Values.global.istioRemote }}
```
Comment (linsun) [inline install/kubernetes/helm/istio/templates/clusterrole.yaml]: is this correct that it only applied to remote?

Comment (linsun) [reply install/kubernetes/helm/istio/templates/clusterrole.yaml]: the other change LG to me.

Comment (morvencao) [reply install/kubernetes/helm/istio/templates/clusterrole.yaml]: Correct, the `clusterrole, clusterrolebinding and serviceaccount` are used for `pilot` to `watch/list` resources in remote cluster, only applied in multicluster topology through the `VPN connectivity`

**Comment by howardjohn** [comment]
@morvencao how important is this? There is no issue/description so I am not sure of the context - is it a security thing, convenience, etc. If it isn't critical should we move this to https://github.com/istio/installer?

**Comment by morvencao** [comment]
@howardjohn 
This is not a critical one. 

I thought end-users don't need these remote cluster resources, which should be deployed only on remote cluster when setting up multi-cluster with VPN Connectivity. However, it's also working to keep these resources there. 

Is there any plan that new installer supports multi-cluster setup with VPN Connectivity, I did't see any resource on remote cluster.

**Pull request closed without merging**